From 3106eb490edc8febc0b6b87878e4d931de5c30f9 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Sun, 14 Dec 2025 19:34:39 +0000
Subject: [PATCH 03/10] CDVDVideoCodecDRMPRIME: Fix AV1 playback using
 v4l2request hwaccel

The FFmpeg AV1 decoder, in contrast to decoders for other codecs, only
call ff_get_format() to initialize a hwaccel when AVCodecContext.pix_fmt
is set to a pix_fmt not supported by the decoder.

Remove the optional initialization of AVCodecContext.pix_fmt to fix AV1
playback using v4l2request hwaccel.

Signed-off-by: Jonas Karlman <jonas@kwiboo.se>
---
 xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp
index 07c5052136..51e77faf19 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp
@@ -343,7 +343,6 @@ bool CDVDVideoCodecDRMPRIME::Open(CDVDStreamInfo& hints, CDVDCodecOptions& optio
     }
   }
 
-  m_pCodecContext->pix_fmt = AV_PIX_FMT_DRM_PRIME;
   m_pCodecContext->opaque = static_cast<void*>(this);
   m_pCodecContext->get_format = GetFormat;
   m_pCodecContext->get_buffer2 = GetBuffer;
-- 
2.34.1

